DataManagementSystem/Agent/RequestOperations/PhysicalRemoval.py

```diff
@@ -30,118 +30,194 @@
-
-  def __call__( self ):
+    DMSRequestOperationsBase.__init__(self, operation, csPath)
+
+  def __call__(self):
     """ perform physical removal operation """
-    bannedTargets = self.checkSEsRSS( access = 'RemoveAccess' )
+
+    # Check whether the ES flag is enabled so we can send the data accordingly.
+    if self.rmsMonitoring:
+      self.rmsMonitoringReporter = MonitoringReporter(monitoringType="RMSMonitoring")
+    else:
+      # # gMonitor stuff
+      gMonitor.registerActivity("PhysicalRemovalAtt", "Physical file removals attempted",
+                                "RequestExecutingAgent", "Files/min", gMonitor.OP_SUM)
+      gMonitor.registerActivity("PhysicalRemovalOK", "Successful file physical removals",
+                                "RequestExecutingAgent", "Files/min", gMonitor.OP_SUM)
+      gMonitor.registerActivity("PhysicalRemovalFail", "Failed file physical removals",
+                                "RequestExecutingAgent", "Files/min", gMonitor.OP_SUM)
+      gMonitor.registerActivity("PhysicalRemovalSize", "Physically removed size",
+                                "RequestExecutingAgent", "Bytes", gMonitor.OP_ACUM)
+
+    bannedTargets = self.checkSEsRSS(access='RemoveAccess')
     if not bannedTargets['OK']:
-      gMonitor.addMark( "PhysicalRemovalAtt" )
-      gMonitor.addMark( "PhysicalRemovalFail" )
+      if self.rmsMonitoring:
+        for status in ["Attempted", "Failed"]:
+          self.rmsMonitoringReporter.addRecord({
+              "timestamp": int(Time.toEpoch()),
+              "host": Network.getFQDN(),
+              "objectType": "File",
+              "operationType": self.operation.Type,
+              "parentID": self.operation.OperationID,
+              "status": status,
+              "nbObject": 1
+          })
+        self.rmsMonitoringReporter.commit()
+      else:
+        gMonitor.addMark("PhysicalRemovalAtt")
+        gMonitor.addMark("PhysicalRemovalFail")
       return bannedTargets
 
     if bannedTargets['Value']:
-      return S_OK( "%s targets are banned for removal" % ",".join( bannedTargets['Value'] ) )
+      return S_OK("%s targets are banned for removal" % ",".join(bannedTargets['Value']))
 
     # # get waiting files
     waitingFiles = self.getWaitingFilesList()
     # # prepare lfn dict
-    toRemoveDict = dict( ( opFile.LFN, opFile ) for opFile in waitingFiles )
+    toRemoveDict = dict((opFile.LFN, opFile) for opFile in waitingFiles)
 
     targetSEs = self.operation.targetSEList
-    gMonitor.addMark( "PhysicalRemovalAtt", len( toRemoveDict ) * len( targetSEs ) )
+
+    if self.rmsMonitoring:
+      self.rmsMonitoringReporter.addRecord({
+          "timestamp": int(Time.toEpoch()),
+          "host": Network.getFQDN(),
+          "objectType": "File",
+          "operationType": self.operation.Type,
+          "parentID": self.operation.OperationID,
+          "status": "Attempted",
+          "nbObject": len(toRemoveDict) * len(targetSEs)
```

Comment:
targetSEs may be None. Anyway, don't bother with that, we are interested only in the number of File objects to treat